### PR TITLE
Implement charge both feature, more accurate cantAfford message

### LINKED ISSUE #5: Charge both XP and money
## What's the enhancement?

Provide a way to charge both XP and money.  

---
Originally requested from @BirkhoffLee on Gitter

### PR DESCRIPTION
**Pull request by BirkhoffLee**
Not tested, ill test it later
close #5

### CODE REVIEW

**Comment by BirkhoffLee** [comment]
The configuration is
```
default: 10.5
pig: 7.25
cow: 0.0
```

Which in 16d4aa64e5181a87bbee768c7fe3f88a5e1647db would be like, charging pig 7.25$ and 7.25 exp which is not desired.

We should make the config like
```
default:
  - money: 10.5
  - exp: 100
pig:
  - money: 7.25
  - exp: 200
cow:
  - money: 0.0
  - exp: 300
```

**Comment by BirkhoffLee** [comment]
Like the config in cddbf5a

**Comment by BirkhoffLee** [comment]
```
[07:22:21 INFO]: [SilkSpawners] Enabling SilkSpawners v3.7.4
[07:22:21 INFO]: [SilkSpawners] WorldGuard was not found and support is disabled
[07:22:21 INFO]: [SilkSpawners] Loading support for v1_11_R1
[07:22:21 INFO]: [SilkSpawners] AutoUpdater is enabled.
[07:22:22 INFO]: [SilkSpawners] Result from AutoUpdater is: NO_UPDATE
[07:22:22 INFO]: [SilkSpawners] BarAPI is disabled due to config setting.
[07:22:22 INFO]: [SilkSpawnersEcoAddon] Enabling SilkSpawnersEcoAddon v2.0.0-SNAPSHOT-b${BUILD_NUMBER}
[07:22:22 ERROR]: Error occurred while enabling SilkSpawnersEcoAddon v2.0.0-SNAPSHOT-b${BUILD_NUMBER} (Is it up to date?)
java.lang.Error: Unresolved compilation problem:
	The method setChargeXP(boolean) is undefined for the type SilkSpawnersEcoAddon

	at de.dustplanet.silkspawnersecoaddon.SilkSpawnersEcoAddon.onEnable(SilkSpawnersEcoAddon.java:138) ~[?:?]
	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:271) ~[spigot-1.11.2.jar:git-Spigot-9ba269b-c19c293]
	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:337) [spigot-1.11.2.jar:git-Spigot-9ba269b-c19c293]
	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:405) [spigot-1.11.2.jar:git-Spigot-9ba269b-c19c293]
	at org.bukkit.craftbukkit.v1_11_R1.CraftServer.enablePlugin(CraftServer.java:376) [spigot-1.11.2.jar:git-Spigot-9ba269b-c19c293]
	at org.bukkit.craftbukkit.v1_11_R1.CraftServer.enablePlugins(CraftServer.java:326) [spigot-1.11.2.jar:git-Spigot-9ba269b-c19c293]
	at net.minecraft.server.v1_11_R1.MinecraftServer.t(MinecraftServer.java:421) [spigot-1.11.2.jar:git-Spigot-9ba269b-c19c293]
	at net.minecraft.server.v1_11_R1.MinecraftServer.l(MinecraftServer.java:382) [spigot-1.11.2.jar:git-Spigot-9ba269b-c19c293]
	at net.minecraft.server.v1_11_R1.MinecraftServer.a(MinecraftServer.java:337) [spigot-1.11.2.jar:git-Spigot-9ba269b-c19c293]
	at net.minecraft.server.v1_11_R1.DedicatedServer.init(DedicatedServer.java:272) [spigot-1.11.2.jar:git-Spigot-9ba269b-c19c293]
	at net.minecraft.server.v1_11_R1.MinecraftServer.run(MinecraftServer.java:544) [spigot-1.11.2.jar:git-Spigot-9ba269b-c19c293]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_121]
[07:22:22 INFO]: Server permissions file permissions.yml is empty, ignoring it
[07:22:22 INFO]: Done (4.217s)! For help, type "help" or "?"
```

File: src/main/resources/config.yml [hunk 1]
```diff
@@ -12,6 +15,12 @@ chargeMultipleAmounts: false
...
-cow: 0.00
+default:
+  - money: 10.5
+  - exp: 100
+pig:
+  - money: 7.25
+  - exp: 200
+cow:
+  - money: 0.0
+  - exp: 20
```
Comment (timbru31) [inline src/main/resources/config.yml]: please add a newline

File: src/main/java/de/dustplanet/silkspawnersecoaddon/listeners/SilkSpawnersEcoSpawnerChangeListener.java [hunk 2]
```diff
@@ -64,24 +68,43 @@ public void onSpawnerChange(SilkSpawnersSpawnerChangeEvent event) {
...
+            if (canAffordXP && canAffordMoney) {
+                plugin.getEcon().withdrawPlayer(player, priceMoney);
+                
+                totalXP -= priceXP;
+                player.setTotalExperience(0);
+                player.setLevel(0);
+                player.giveExp(totalXP);
+                player.sendMessage(ChatColor.translateAlternateColorCodes('&', plugin.getConfig().getString("afford")).replace("%money%", Double.toString(price)));
+            } else {
+                if (!canAffordXP)
```
Comment (timbru31) [inline src/main/java/de/dustplanet/silkspawnersecoaddon/listeners/SilkSpawnersEcoSpawnerChangeListener.java]: please use brackets for if/else

File: src/main/java/de/dustplanet/silkspawnersecoaddon/listeners/SilkSpawnersEcoSpawnerChangeListener.java [hunk 3]
```diff
@@ -64,24 +68,43 @@ public void onSpawnerChange(SilkSpawnersSpawnerChangeEvent event) {
             plugin.getPendingConfirmationList().remove(playerName);
         }
 
+        int totalXP = player.getTotalExperience();
+        
         if (plugin.isChargeXP()) {
```
Comment (timbru31) [inline src/main/java/de/dustplanet/silkspawnersecoaddon/listeners/SilkSpawnersEcoSpawnerChangeListener.java]: please refactor/extract these if else block into multiple functions, (was way too long before anyways)

File: src/main/java/de/dustplanet/silkspawnersecoaddon/listeners/SilkSpawnersEcoSpawnerChangeListener.java [hunk 4]
```diff
@@ -64,24 +68,43 @@ public void onSpawnerChange(SilkSpawnersSpawnerChangeEvent event) {
...
                 player.setLevel(0);
                 player.giveExp(totalXP);
                 player.sendMessage(ChatColor.translateAlternateColorCodes('&', plugin.getConfig().getString("afford")).replace("%money%", Double.toString(price)));
             } else {
-                player.sendMessage(ChatColor.translateAlternateColorCodes('&', plugin.getConfig().getString("cantAfford")));
+                player.sendMessage(ChatColor.translateAlternateColorCodes('&', plugin.getConfig().getString("cantAffordXP")));
+                event.setCancelled(true);
+            }
+        } else if (plugin.isChargeBoth()) {
+            Boolean canAffordXP = totalXP >= priceXP;
```
Comment (timbru31) [inline src/main/java/de/dustplanet/silkspawnersecoaddon/listeners/SilkSpawnersEcoSpawnerChangeListener.java]: `boolean`

**Comment by BirkhoffLee** [comment]
Sorry, I'm very busy at school now, and I'm not actually familiar with Java. But my changes are clear, maybe you can change them by yourself? Thanks!

**Comment by timbru31** [comment]
Will do, thanks for the start though :)

**Comment by BirkhoffLee** [comment]
You're welcome, and thanks for the awesome plugin too! :)

**Comment by timbru31** [comment]
Closed in favor of #9

**Pull request closed without merging**